dotfiles/run.sh

```diff
@@ -0,0 +1,12 @@
+DOTFILES_URI="${DOTFILES_URI}"
+ROOT="${ROOT}"
+
+if [ -n "$${DOTFILES_URI// }" ]; then
+    echo "✨ Applying dotfiles for user $USER"
+    coder dotfiles "$DOTFILES_URI" -y 2>&1 | tee -a ~/.dotfiles.log
+
+    if [ "$ROOT" = "true" ]; then
+        echo "✨ Applying dotfiles for user root"
+        sudo $(which coder) dotfiles "$DOTFILES_URI" -y 2>&1 | tee -a ~/.dotfiles-root.log
```

Comment:
I didn't do it because `tee` is getting it's output from sudo, not from the actual coder CLI, which means that `tee` is not running as root and therefore can't write in `/root`.

We could do something like `sudo sh -c "$(which coder) dotfiles \"$DOTFILES_URI\" -y 2>&1 | tee -a /root/.dotfiles.log"` but I don't know if it matters that much, what do you think ?

Reply:
Probably yes. I don't think it's a common use case. So won't affect many people.

Reply:
given how this PR is going with custom users rather than arbitrary root, I will make it so that it dynamically resolves the user's home and write to it